src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -242,6 +245,34 @@ public static PackageSpec GetPackageSpec(IEnumerable<IMSBuildItem> items)
             return result;
         }
 
+        /// <summary>
+        /// Remove missing project dependencies. These are typically caused by
+        /// non-NuGet projects which are missing the targets needed to walk them.
+        /// Visual Studio ignores these projects so from the command line we should
+        /// also. Build will fail with the appropriate errors for missing projects
+        /// restore should not warn or message for this.
+        /// </summary>
+        public static void RemoveMissingProjects(DependencyGraphSpec graphSpec)
+        {
+            var existingProjects = new HashSet<string>(
+                graphSpec.Projects.Select(e => e.RestoreMetadata.ProjectPath),
+                PathUtility.GetStringComparerBasedOnOS());
+
+            foreach (var project in graphSpec.Projects)
+            {
+                foreach (var framework in project.RestoreMetadata.TargetFrameworks)
+                {
+                    foreach (var projectReference in framework.ProjectReferences.ToArray())
```

Comment:
nvm, didn't see the remove call below

---
In reply to: [114226717](https://github.com/NuGet/NuGet.Client/pull/1350#discussion_r114226717) [](ancestors = 114226717)